src/main/java/com/bathanh/exercise/basic/string/Ex306.java

```diff
@@ -0,0 +1,26 @@
+package com.bathanh.exercise.basic.string;
+
+public class Ex306 {
+
+    public int calculateBarcode(final String n) {
+        if (n.length() != 12) {
+            throw new IllegalArgumentException();
+        }
+        int verify = 0;
```

Comment:
Nhớ quy tắc khai báo biến thì phải minimal scope của biến. Như chương trình này thì biến nên được khai báo ngay trước vòng for. Nhưng thật ra biến này có thể bỏ

Reply:
solved